PokeOneWeb.Data/Entities/Interfaces/IEntity.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace PokeOneWeb.Data.Entities.Interfaces
+{
+    /// <summary>
+    /// Common interface for all entities.
+    /// </summary>
+    public interface IEntity
```

Comment:
Good question, here's some reasons I prefer an interface over an abstract class in this case:
- Interface is mainly used as generic type constraint for classes which are injected through dependency injection. Thus it is mentioned in the respective interface for the dependency (i.e. IRepository) --> avoid use of custom classes in interfaces.
- General best practice of reducing inheritance, not a lot of duplicated code would be avoided by introducing an abstract class.
- Inheritance in combination with Entity Framework can lead to unexpected results / makes things more complicated
- Preference of having all properties of a POCO / entity type in the same file instead of split across multiple files
- Lines up with the other, disjunct entity interfaces INamedEntity and IHashedEntity, which can be used mutually exclusively or both (would not be possible with inheritance)
- Similar to Marker interface pattern: https://en.wikipedia.org/wiki/Marker_interface_pattern